oracle-example/service/src/main/kotlin/net/corda/examples/oracle/service/service/Oracle.kt

```diff
@@ -32,8 +43,12 @@ class Oracle(val services: ServiceHub) : SingletonSerializeAsToken() {
 
     // Returns the Nth prime for N > 0.
     fun query(n: Int): Int {
-        require(n > 0) { "n must be at least one." } // URL param is n not N.
-        return primes.take(n).last()
+        return cache.get(n) ?: {
```

Comment:
`return cache.get(n) ?: cache.put(n, primes.take(n).last())!!`
Could look neater if you want to get rid of the `require`